core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java

```diff
@@ -405,4 +405,92 @@ default RelDataType deriveDecimalModType(RelDataTypeFactory typeFactory,
     return null;
   }
 
+  /**
+   * Infers the return type of a decimal truncate operation. Decimal truncate
+   * involves at least one decimal operand.
+   *
+   * <p>The default implementation is SQL:2003 compliant: the declared type of
+   * the result is the declared type of the first operand (decimal to be truncated).
+   *
+   * @see Glossary#SQL2003 SQL:2003 Part 2 Section 6.27
```

Comment:
I know SQL standard document the general operators (`+`, `-`, `*`, `/`, `%`) for decimal types. But can we find the guideline in SQL standard for `truncate` and `round`?

Reply:
There is not a consistent one from what we found SQL Server seems to go with what calcite does where as Oracle does not.
Also i think that is outside the scope of this PR..

Reply:
My point is the document added in this PR. If it is not a standard manner, we'd better not add this confusing paragraph.

> The default implementation is SQL:2003 compliant: the declared type of
> the result is the declared type of the first operand (decimal to be truncated).
>  @see Glossary#SQL2003 SQL:2003 Part 2 Section 6.27